prometheus_fzj_weather_exporter/main.py

```diff
@@ -1,58 +1,50 @@
 #!/usr/bin/env python3
 # This file is licensed under the ISC license.
 # Oskar Druska 2022
-# For further information look up LICENSE.txt
-
-# exporter entry point
-
-# test usage:
-# > python prometheus_fzj_weather_exporter/main.py --web.listen-address 127.0.0.1:9184
-# > curl 127.0.0.1:9184 (in different window)
-# expected output (similar to):
-# > # HELP fzj_weather_air_temperature temperature in celsius
-# > # TYPE fzj_weather_air_temperature gauge
-# > fzj_weather_air_temperature 14.0
-# (equivalent output for other data i.e. humidity)
+# For further information look up LICENSE
 
 import sys
 import argparse
+from argparse import RawTextHelpFormatter
 import time
 from prometheus_client import start_http_server, REGISTRY
 from prometheus_fzj_weather_exporter import exporter_file
 
 
 def main():
     args = get_parsed_args()
-
+    url = "https://www.fz-juelich.de/de/gs/ueber-uns/meteo/aktuelle-wetterdaten/wetterdaten"
     try:
-        REGISTRY.register(exporter_file.FZJWeatherExporter(args.insecure))
+        REGISTRY.register(exporter_file.FZJWeatherExporter(url, args.insecure))
     except ConnectionError as c:
         sys.exit(c.strerror)
 
-    if args.listenaddress is None:
-        start_http_server(port=9184, addr='127.0.0.1')
-    else:
+    # start the http server
+    if args.listenaddress:
         ip, port = args.listenaddress.split(":")
         if ip:
             start_http_server(port=int(port), addr=ip)
         else:  # listen on all interfaces
             start_http_server(port=int(port))
+    else:
+        start_http_server(port=9184, addr='127.0.0.1')
 
-    # keep the thing going indefinitely
+    # keep the exporter running indefinitely
     while True:
         time.sleep(1)
 
 
 def get_parsed_args():
     parser = argparse.ArgumentParser(
-        description='Set up the Prometheus exporter (connection ports)')
+        description='Set up the Prometheus exporter (connection ports)',
+        formatter_class=RawTextHelpFormatter)
     group = parser.add_argument_group()
     group.add_argument(
         '-w', '--web.listen-address',
         type=str,
         dest='listenaddress',
         help='Address and port to expose metrics and web interface. Default: ":9184"\n'
-             'To listen on all interfaces, omit the IP. ":<port>"\n'
+             'To listen on all interfaces, omit the IP: ":<port>"\n'
```

Comment:
IIRC, this behavior recently changed... 

See https://github.com/psyinfra/prometheus-eaton-ups-exporter/commit/162c390919ca34a850831a813e4cd6afda7c800a

Reply:
Dang it, I forgot about that.
I would follow the eaton-ups-exporter and change it to `0.0.0.0`